From f8af9062980c4ead9381af7cb5ec9e6d4d993f77 Mon Sep 17 00:00:00 2001
Message-Id: <f8af9062980c4ead9381af7cb5ec9e6d4d993f77.1645215914.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Fri, 18 Feb 2022 21:23:01 +0100
Subject: [PATCH] [web] Use relative paths in xhr requests

This allows to prefix the path e.g. when using a reverse proxy.
---
 src/web/web-service/frontend/res/js/app.js | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/web/web-service/frontend/res/js/app.js b/src/web/web-service/frontend/res/js/app.js
index 6a2959fb7..870520074 100644
--- a/src/web/web-service/frontend/res/js/app.js
+++ b/src/web/web-service/frontend/res/js/app.js
@@ -130,7 +130,7 @@
             $scope.menu[index].show = true;
             if (index == 1) {
                 $scope.isLoading = true;
-                $http.get('/available_network').then(function(response) {
+                $http.get('available_network').then(function(response) {
                     $scope.isLoading = false;
                     if (response.data.error == 0) {
                         $scope.networksInfo = response.data.result;
@@ -140,7 +140,7 @@
                 });
             }
             if (index == 3) {
-                $http.get('/get_properties').then(function(response) {
+                $http.get('get_properties').then(function(response) {
                     console.log(response);
                     if (response.data.error == 0) {
                         var statusJson = response.data.result;
@@ -257,7 +257,7 @@
                 };
                 var httpRequest = $http({
                     method: 'POST',
-                    url: '/join_network',
+                    url: 'join_network',
                     data: data,
                 });
 
@@ -277,7 +277,7 @@
 
             $scope.qrcode = function() {
                 $scope.isLoading = false;
-                $http.get('/get_qrcode').then(function(response) {
+                $http.get('get_qrcode').then(function(response) {
                     console.log(response);
                     $scope.res = response.data.result;
                     if (response.data.result == 'successful') {
@@ -323,7 +323,7 @@
                 $scope.isForming = true;
                 var httpRequest = $http({
                     method: 'POST',
-                    url: '/form_network',
+                    url: 'form_network',
                     data: data,
                 });
 
@@ -371,7 +371,7 @@
                 };
                 var httpRequest = $http({
                     method: 'POST',
-                    url: '/add_prefix',
+                    url: 'add_prefix',
                     data: data,
                 });
 
@@ -397,7 +397,7 @@
                 };
                 var httpRequest = $http({
                     method: 'POST',
-                    url: '/delete_prefix',
+                    url: 'delete_prefix',
                     data: data,
                 });
 
@@ -416,7 +416,7 @@
             };
             var httpRequest = $http({
                 method: 'POST',
-                url: '/commission',
+                url: 'commission',
                 data: data,
             });
             
-- 
2.35.1

